### ISSUE TITLE
"Affordance" in EPUB A11Y

### ISSUE THREAD

**Issue by murata2makoto** [mattgarrish/epub-revision#1961]
While trying to create a JIS standard for EPUB Accessibility, I stumbled over this word.  It appears in [2.1](https://www.w3.org/TR/epub-a11y-11/#sec-disc-intro) of EPUB Accessibility 1.1.

From [an article in Wikipedia](https://en.wikipedia.org/wiki/Affordance), this term in the context of Human–Computer Interaction appears to mean suggestions to users.  But our metadata are not suggestions.  They are clear indications of accessibility characteristics.  I thus propose to avoid this term.

**Comment by mattgarrish** [comment]
Affordances do offer action possibilities, but I think bringing HCI into the discussion is going a bit far afield. We're not discussing the ways in which the affordances are presented to the user to interact with, only that they exist.

And, technically, we are only suggesting ways of interacting with the content by listing the affordances. That I put alt text in a book doesn't mean you have to read it, or that you have to read it visually. Its presence suggests different ways for users to process the information, and they'll use the method they're most used to.

I think in this case, though, we can probably rely on the general definition of an affordance:

> the quality or property of an object that defines its possible uses or makes clear how it can or should be used

https://www.merriam-webster.com/dictionary/affordance

The alternative is to use "features". It's doable, but doesn't come off as precise.

**Comment by murata2makoto** [comment]
The Wikipedia article makes clear that this term was coined by James J. Gibson.  His original definition is certainly different from the one you quoted.    Why do we have to rely on an uncommon definition?  Apparently, there have been lots of confusion about this term.  I do not think that this word is strongly needed in our spec.

**Comment by mattgarrish** [comment]
> His original definition is certainly different from the one you quoted.

How so? The very first line introducing the quote is:

> Affordance is what the environment offers the individual.

And Gibson is cited in the dictionary definition I posted.

You appear to be referring to this part:

> In 1988, Donald Norman appropriated the term affordances in the context of Human–Computer Interaction to refer to just those action possibilities that are readily perceivable by an actor. ... Gibson's original definition of affordances allows that the actor may throw the chair and sit on the ball, because this is objectively possible.

**Comment by mattgarrish** [comment]
But if you have a better term for the affordances added by the author, could you please provide it?

**Comment by lordt4ever** [comment]
But me I wasn't changed make it for me

**Comment by lordt4ever** [comment]
Sen it

**Comment by murata2makoto** [comment]
I wrote:

>His original definition is certainly different from the one you quoted.

Matt wrote:
>How so? The very first line introducing the quote is:
>
>Affordance is what the environment offers the individual.

The original definition by Gibson does not require that the environment makes clear to the user what it offers.   But making clear to the user is the whole point of Norman's definition in the context of HCI.  Our metadata certainly makes clear what is possible.  Furthermore, our metadata is not a suggestion.  It is a very explicit manifestation.  I thus think affordance is not the right word here.

How about this?

Each user must be able to gauge the usability of an EPUB Publication when he purchases, borrows, or otherwise obtains it, a determination that requires knowing how accessible the EPUB publication is to the user.

**Comment by clapierre** [comment]
We can tweak what the definition of "accessMode" and "accessModeSufficient" are but the actual metadata should not be affected as these terms have been accepted by Schema.org and took years to get into their specification.

**Comment by mattgarrish** [comment]
> The original definition by Gibson does not require that the environment makes clear to the user what it offers.

Sure, and neither does an EPUB. You're effectively making the point of the sentence. The affordances are what are available. The metadata identifies them. Without the metadata, you don't know what affordances are available until you access the publication and discover them.

But there's also another use of the term in the accessModeSufficient definition:

> (i.e., unlike access modes, this property takes into account any affordances for content that is not broadly accessible, such as the inclusion of transcripts for audio content).

**Comment by murata2makoto** [comment]
@clapierre 
Yes, I am just proposing wordsmithing.

@mattgarrish 
I continue to think that the use of this word makes this spec ambiguous and incomprehensive without any advantages.

**Comment by murata2makoto** [comment]
>But there's also another use of the term in the accessModeSufficient definition:

>>(i.e., unlike access modes, this property takes into account any affordances for content that is not broadly accessible, such as the inclusion of transcripts for audio content).

I do not understand this parenthesized sentence well.  In particular, I do not understand the relationship between it and the preceding sentence (" An EPUB Publication can have more than one set of sufficient access modes for its consumption depending on the types of content it includes").

I am asked by JIS to either translate the word affordance or add a note.  I am at a loss.

**Comment by mattgarrish** [comment]
The difference is that access modes only describe the nature of the content. If you have access modes of textual and visual, all you know is that there is text and visual content, so by default you need to be able to read text and view the images.

Sufficient access modes describe all the pathways for reading the content. One set is always the sum of the access modes. The others account for the affordances, like alt text and descriptions. If those affordances provide the same information, then there is a pathway for users who cannot consume the content in its default form.

Affordances are all the alternatives (for lack of a better word) provided to allow reading beyond the default nature of the content -- alt text, long descriptions, transcripts, synchronized audio, etc.

If it's that problematic for translation, though, perhaps we can rewrite the sentence in another way. Maybe:

> An EPUB Publication can have more than one set of sufficient [access modes](http://localhost:120/a11y/#confreq-schema-accessMode) depending on the alternatives provided to enable reading in another mode. For example, if alternative text and descriptions are provided for all the images in a publication, it would have both its default textual and visual sufficient access mode _and_ a purely textual sufficient access mode.

Alternatives isn't quite accurate, but maybe is good enough.

**Comment by GeorgeKerscher** [comment]
Thinking of words that can make this more clear, would "available features" be helpful? I am thinking that these affordances are available features in the publication. Available functionality may be another translation that would work.

**Comment by murata2makoto** [comment]
In a JIS meeting yesterday, we decided to wait for the resolution of this issue.

**Comment by clapierre** [comment]
I think this proposed text from Matt works as well, especially if it is easier to translate/understand by others.



If it's that problematic for translation, though, perhaps we can rewrite the sentence in another way. Maybe:

An EPUB Publication can have more than one set of sufficient access modes<http://localhost:120/a11y/#confreq-schema-accessMode> depending on the alternatives provided to enable reading in another mode. For example, if alternative text and descriptions are provided for all the images in a publication, it would have both its default textual and visual sufficient access mode and a purely textual sufficient access mode.

Charles

**Comment by iherman** [comment]
The issue was discussed in a [meeting](https://www.w3.org/publishing/groups/epub-wg/Meetings/Minutes/2022-02-17-epub#section4) on 2022-02-17

- no resolutions were taken

<details><summary><i>View the transcript</i></summary>

### 4. Wording of Affordance.

_See github issue [epub-specs#1961](https://github.com/w3c/epub-specs/issues/1961)._




**Wendy Reid:** i think the only question is, Makoto, is it okay to go with the last suggested text from mgarrish on the issue?.  

**Murata Makoto:** different people have different understanding of the word affordance, so let's use something else.  

**Wendy Reid:** I think mgarrish suggested some form of "alternatives".  

**Murata Makoto:** George suggested "available features".  

**Wendy Reid:** mgarrish suggested that in the specific text we're talking about "alternatives" might work.  

> An EPUB Publication can have more than one set of sufficient access modes depending on the alternatives provided to enable reading in another mode..

**Dave Cramer:** I like that wording.  

**Murata Makoto:** there are two instances of the word "affordance". Will we fix both?.  

**Wendy Reid:** the other is in the explanation for `accessModeSufficient`.  

> *Wendy Reid:* "this property takes into account any affordances for content that is not broadly accessible".

**Wendy Reid:** so yes, I think using "alternatives" here makes sense too.  

**Murata Makoto:** thank you!.  

</details>

**Comment by iherman** [comment]
Created issue #2000. This issue will be closed if and when that PR is merged.
